### PR TITLE
Final Review

### PR DESCRIPTION
**Pull request by iLynette**
## Project Requirements
Splash screen

- A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- Sign up and log in pages

- The user should be able to register in the app with full name, email and password (all mandatory).

- The user can log into the app using email and password.

- If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

Home page (catagories page)

- When the user logs in, they are presented with the categories page.

- For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.

- When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.

- There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

Transactions page

- For a given category, the list of transactions is presented, ordered by the most recent.

- At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).

- There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.

- When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page

- The user fills out a form to create a new category, indicating their name and icon (both mandatory).

- The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.

- When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page

- The user fills out a form to create a new transaction with:

- name (mandatory)

- amount (mandatory)

- categories (mandatory at least one)

- The user click (or taps) the "Save" button to create the new category, and is taken to the transactions page for that category.

- When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

Thank you for the review ☺️

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,93 @@
...
+Give a ⭐️ if you like this project!
+
+## Acknowledgments
+
+- Thanks [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) for the original design idea.
+- Thanks to my morning session partners, coding partners and teammates.
+
+## 📝 License
+
+This project is [MIT](./MIT.md) licensed.
```
Comment (ahmetbozaci) [inline README.md:93]: - [ ] Kindly fix the license link to make a professional readme. It is broken.
- [ ] Kindly add creative common license. ### Optıonal

File: README.md [hunk 2]
```diff
@@ -1,24 +1,93 @@
...
+- Ruby
+- Ruby on Rails
+- PostgreSQL
 
-* Services (job queues, cache servers, search engines, etc.)
+## Live Demo 
 
-* Deployment instructions
+- [Check the app](https://stark-island-46203.herokuapp.com/)
+- [Video Presentation](https://www.loom.com/share/3d5c1263e3954ca5a641a0469476e0cc)
```
Comment (ahmetbozaci) [inline README.md:23]: - Well done with the presentation video to a demo of the website. Kindly show some codes that you are proud of it. You are telling in the video but you are not showing any code. [Requirement](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#:~:text=you%20should%20also%20highlight%20some%20interesting%20piece%20of%20code%20or%20something%20you%20built%20that%20you%20are%20very%20proud%20of.)
_You can create another video about the code and combine them._

File: app/views/categories/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      <div>
+        <p>$<%= payment.amount %></p>
+      </div>
+    </div>
+  </div> 
+<% end %>
+
+<div class="payment-btn">
+  <%= link_to "ADD NEW TRANSACTION", new_payment_path(category_id: @category.id), class: 'expense-btn' %>
+</div>
```
Comment (ahmetbozaci) [inline app/views/categories/show.html.erb:42]: - [ ] I can see other users' categories and transactions. Every user should see their category and transactions. Kindly fix it.
_I create `test` user and `test2` user. Test2 user can see test user's category._
image

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# ♻️♻️♻️ STATUS: Changes Required  ♻️♻️♻️


 @iLynette  ✋🏼 

#### Your project is excellent. You are almost done. However, there are some issues you should fix. Please fix them then your project will be approved.

___ 

### 💯  Nice to have 💯
- Your git-flow is correct. ✔️
- You set up linter files and no errors. ✔️
- Your pull request is very professional. It has a meaningful title and summary. Keep it up.✔️
- I can't log in if ı am not signup for the website.✔️
- You follow the design guidelines(color, layout, typography)✔️
- You create tests and all tests are passed✔️
- I can see the list of transactions ordered by the most recent and ı can see the total amount on the top.✔️
___


### ♻️ Request changes ♻️

- [ ] _Check the comments under the review._

___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏
almost
almost


___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._


___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

File: README.md (hunk 2 above)
Comment (iLynette) [reply README.md:23]: Thank you for your review but the instruction this was the instruction `you should also highlight some interesting piece of code **or** something you built that you are very proud of.` I had the choice to either show a piece of code, or mention something that I am proud of, which I believe I did.

File: app/views/categories/show.html.erb (hunk 3 above)
Comment (iLynette) [reply app/views/categories/show.html.erb:42]: Thank you, I have fixed this

File: README.md (hunk 1 above)
Comment (iLynette) [reply README.md:93]: Thank you, I added the MIT licence but the creative commons licence was already there. @ahmetbozaci

Comment (ahmetbozaci) [reply README.md:93]: @iLynette  I didn't see any [creative common ](https://creativecommons.org/licenses/by-nc/4.0/)license in the readme file. 
image

File: README.md (hunk 2 above)
Comment (ahmetbozaci) [reply README.md:23]: @iLynette You are right. Kindly ignore it.

File: README.md (hunk 1 above)
Comment (iLynette) [reply README.md:93]: The creative common licence requires that I give credit to the original author, and I gave credit to Gregoire Vella on Behance? is there more to it? Please clarify @ahmetbozaci

Comment (ahmetbozaci) [reply README.md:93]: @iLynette Like adding another license([creative common license](https://creativecommons.org/licenses/by-nc/4.0/)) in the license section. But ı make it optional.

**Review by Cathella — APPROVED** [review]
## 🔥 Status: Approved 🟢

### Hi @iLynette 👋🏻, 

Good job making the requested changes 💪🏻!
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

rosycheeks-mochi-peach


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by iLynette** [comment]
Thank you @Cathella

**Pull request merged**